### ISSUE TITLE
[Suggestion] UnrealEd code editor new features

### ISSUE THREAD

**Issue by an-eternity** [OldUnreal/UnrealTournamentPatches#845]
### Issue description:
Currently UnrealEd code editor is in state that is barely enough to work with huge amount of scripts effectively. It misses a lot of very useful functionality - from such simple as Line Numbering to such complex as Code Completion. This makes it reasonable to use external code editors instead in order to save the time, where lack of integration with UC environment brings another issues though...

### Suggestion:
Here is a list of possible improvements that could make UnrealEd code editor more usable.
1. Code completion.
   1.1. Within current script context.
   1.2. Within context of all parent Classes scripts (if possible).
        1.2.1. Within context of related/loaded Classes scripts (if possible).
        1.2.2. With appropriate note which Class an item is related to.
   1.3. Support for Structs and Enums.
   1.4. "Goto Definition/Declaration/Implementation" option in context menu and via hotkey "Ctrl+MouseLeftButton".
2. Line numbering (issue #717).
3. Show `defaultproperties` field for scripts loaded from packages (if possible - with properly decoded Enum names).
4. List of functions, global variables and constants of the current Class for quick access.
   4.1. (If item 1.2. implemented) Additional list of inherited functions, global variables and constants with appropriate note which parent Classes they are related to.
5. Syntax dynamic highlighting.
   5.1. Support multiline dynamic highlighting (if possible).
        5.1.1. Matching brackets dynamic highlighting (scope highlighting).
   5.2. Matching selected text highlighting.
6. Customizable color settings.
7. "Find and Replace" dialog (with option to auto-replace all occurrences).

More suggestions can be added further...
If somebody has a suggestion that could improve UnrealEd code editor more and is not in the list yet - add it here in comments...

* This list is rather an approximate 'guideline', items it contains are not an imperative, it may appear practically impossible or not reasonable to implement some of them. If at least some of them implemented, it should make code editor more useful/effective in some extent. If all items implemented, it would be the best case, of course.

**Comment by SeriousBuggie** [comment]
Need add button for return back to previous cursor location, after jump by click on some name with CTRL.

Use case: you read code, can use CTRL + click for look implementation of item, and later can press back button (which also bind to button `Previous` for mouse - MK_XBUTTON1), for return back to your place.

Initially work inside one class. Later can be extended to work cross classes.

**Comment by an-eternity** [comment]
Yes, return back is useful function. especially when working with huge scripts and when Jump follows to another classes...
Maybe Bookmarks would also be useful (i use it often when working with UC in external IDE).

**Comment by SeriousBuggie** [comment]
#630 also useful thing which can be added in this list.

**Comment by SeriousBuggie** [comment]
And #631 too.

**Comment by SeriousBuggie** [comment]
#887 can be added to list.

**Comment by an-eternity** [comment]
> #887 can be added to list.

Added... Item 2.3.
I am not quite sure how this should work and look like... Is this an additional column, like the line numbering, that by result of the compilation will show a bytecode offset numbers towards each line that corresponds to something in the bytecode?
Or, probably, it could an extension to Search functionality (that resolves a given bytecode offset number into the script line number), or could be shown via Hints from the mouse cursor/pointer...
